pokedex/model/handwritten/float_api.asl

```diff
@@ -0,0 +1,158 @@
+// float_api.asl act as glue file to perform calculation and data type
+// conversion between ASL model and softfloat model.
+//
+// This file contains only *pure* softfloat operation: functions are operate
+// with only pure value in and pure value out, no ASL internal architecture
+// states will be modified during function execution (softfloat states might be
+// changed though).
+
+let CANONICAL_NAN : bits(32) = (0x7fc0_0000)[31:0];
+
+func f_is_not_a_number(v : bits(32)) => boolean
+begin
+  return v[31] == '1' && IsOnes(v[30:21]) && !(IsZero(v[20:0]));
+end
```

Comment:
This function is complete wrong. NaN check is not related to the sign, and F32 has 8 exp bits and 23 sig bits.